Nodes/puzzle_example/coap_client.c

```diff
@@ -0,0 +1,234 @@
+/*
```

Comment:
crazy idea: make this coap client a module on its own :)

Reply:
That file exists two times in the RIOT repository - one could try to move it to sys/shell/commands upstream permanently. But would ofc depend on the community approval......aaaand your approval, as it could take much longer than just refactoring it around in our little project here ;)

Reply:
~~May I remind you that last time you had a "crazy idea" on one of mine PRs, it resulted in couple hours of work and a mail at security@riot? 😘~~